src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -25,131 +18,72 @@ HOSTFXR_UTILITY::~HOSTFXR_UTILITY()
+        PCWSTR              pcwzArguments,
+        HANDLE              hEventLog,
+        _Inout_ STRU*		pStruHostFxrDllLocation,
+        _Out_ DWORD*		pdwArgCount,
+        _Out_ BSTR**		ppwzArgv
 )
 {
-    HRESULT             hr = S_OK;
-    STRU                struDllPath;
-    STRU                struArguments;
-    STRU                struHostFxrPath;
-    STRU                struRuntimeConfigLocation;
-    DWORD               dwPosition;
-
-    // Obtain the app name from the processPath section.
-    if (FAILED(hr = struDllPath.Copy(pwzExeAbsolutePath)))
-    {
-        goto Finished;
-    }
+    HRESULT hr = S_OK;
 
-    dwPosition = struDllPath.LastIndexOf(L'.', 0);
-    if (dwPosition == -1)
-    {
-        hr = E_FAIL;
-        goto Finished;
-    }
+    const fs::path exePath(pwzExeAbsolutePath);
 
-    hr = UTILITY::ConvertPathToFullPath(L".\\hostfxr.dll", pcwzApplicationPhysicalPath, &struHostFxrPath);
-    if (FAILED(hr))
+    if (!exePath.has_extension())
     {
-        goto Finished;
+        return false;
     }
 
-    struDllPath.QueryStr()[dwPosition] = L'\0';
-    if (FAILED(hr = struDllPath.SyncWithBuffer()))
-    {
-        goto Finished;
-    }
+    const fs::path physicalPath(pcwzApplicationPhysicalPath);
+    const fs::path hostFxrLocation = physicalPath / L"hostfxr.dll";
 
-    if (!UTILITY::CheckIfFileExists(struHostFxrPath.QueryStr()))
+    if (!exists(hostFxrLocation))
     {
-        // Most likely a full framework app.
-        // Check that the runtime config file doesn't exist in the folder as another heuristic.
-        if (FAILED(hr = struRuntimeConfigLocation.Copy(struDllPath)) ||
-            FAILED(hr = struRuntimeConfigLocation.Append(L".runtimeconfig.json")))
-        {
-            goto Finished;
-        }
-        if (!UTILITY::CheckIfFileExists(struRuntimeConfigLocation.QueryStr()))
-        {
+        fs::path runtimeConfigLocation = exePath;
+        runtimeConfigLocation.replace_extension(L".runtimeconfig.json");
 
-            hr = E_APPLICATION_ACTIVATION_EXEC_FAILURE;
-            UTILITY::LogEventF(hEventLog,
-                EVENTLOG_ERROR_TYPE,
-                ASPNETCORE_EVENT_INPROCESS_FULL_FRAMEWORK_APP,
-                ASPNETCORE_EVENT_INPROCESS_FULL_FRAMEWORK_APP_MSG,
-                pcwzApplicationPhysicalPath,
-                hr);
-        }
-        else
+        if (!exists(runtimeConfigLocation))
         {
-            // If a runtime config file does exist, report a file not found on the app.exe
-            hr = HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
-            UTILITY::LogEventF(hEventLog,
-                EVENTLOG_ERROR_TYPE,
-                ASPNETCORE_EVENT_APPLICATION_EXE_NOT_FOUND,
-                ASPNETCORE_EVENT_APPLICATION_EXE_NOT_FOUND_MSG,
-                pcwzApplicationPhysicalPath,
-                hr);
+            EVENTLOG(hEventLog, INPROCESS_FULL_FRAMEWORK_APP, pcwzApplicationPhysicalPath, 0);
+            return E_FAIL;
```

Comment:
If we are going to start using std::filesystem, we should start handling exceptions. Next PR?

Reply:
I hope they would be handled by global catch from other PR. Not much we can do otherwise.